src/librustc/session/mod.rs

```diff
@@ -158,6 +158,9 @@ impl Session {
     pub fn fatal(&self, msg: &str) -> ! {
         panic!(self.diagnostic().fatal(msg))
     }
+    pub fn fatal_panic_only(&self) -> ! {
+        panic!(errors::FatalError)
```

Comment:
We should just `panic!()` I think, passing `errors::FatalError` has special meaning. In fact maybe it doesn't make sense to call this when we're already aborting and we should just `panic!`